spec/lib/html_processors/amp_img_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'html_processor'
+require 'html_processor/amp_img'
+
+RSpec.describe HTMLProcessor::AmpImg do
+  let(:processor) { described_class.new(html) }
+
+  describe '.process' do
+    subject { Nokogiri::XML(processed_html.strip).children.first }
+    let(:attributes) { subject.attributes.transform_values!(&:value) }
+    let(:processed_html) { processor.process(HTMLProcessor::IMG) }
+
+    let(:xpath) do
+      '//img'
+    end
+
+    it 'creates an amp-img tag' do
+      expect(subject.name).to eq('amp-img')
+    end
+
+    let(:html) do
+      <<-EOHTML
+          <img
+              srcset="test_srcset"
+              src="test_src"
+              sizes="test_sizes"
+              alt="test_alt"
+              attribution="test_attribution"
+              height="test_height"
+              width="test_width" />
+        EOHTML
+    end
+
+    it 'copies the attributes across correctly from the original iframe' do
+      expect(attributes).to include(
+        'srcset' => 'test_srcset',
+        'src' => 'test_src',
+        'sizes' => 'test_sizes',
+        'alt' => 'test_alt',
+        'attribution' => 'test_attribution',
+        'height' => 'test_height',
+        'width' => 'test_width'
+      )
+    end
+
+    describe 'supporting javascript disabled' do
+      let(:amp_img_tag) { Nokogiri::XML(processed_html.strip).children.first }
+      subject { amp_img_tag.children.first }
+      let(:noscript_tag) { subject.children.first }
+      let(:inner_attributes) { noscript_tag.attributes.transform_values(&:value) }
```

Comment:
re line length, please refer to the line length comment on https://github.com/moneyadviceservice/rad/pull/416 lib/adviser_list_csv.rb